### PR TITLE
C#: Summarized callable

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR, we
- Fix issue with the isAutoGenerated predicate (contains to much).
- Fix issue with dataflow library such that only relevant summarized callables will be used.
- Update tests and dependencies to other languages.

### CODE REVIEW

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -743,11 +744,11 @@ private module Cached {
       FlowSummaryImpl::Public::SummarizedCallable c,
       FlowSummaryImpl::Private::SummaryNodeState state
     ) {
-      useFlowSummary(c) and
+      c instanceof DataFlowSummarizedCallable and
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll:747]: You may as well change the type of `c` to be `DataFlowSummarizedCallable`.

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll [hunk 2]
```diff
@@ -743,11 +744,11 @@ private module Cached {
...
       FlowSummaryImpl::Private::summaryNodeRange(c, state)
     } or
     TSummaryParameterNode(FlowSummaryImpl::Public::SummarizedCallable c, ParameterPosition pos) {
-      useFlowSummary(c) and
+      c instanceof DataFlowSummarizedCallable and
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll:751]: Same as above.

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll:747]: Oh right!

**Comment by michaelnebel** [comment]
None of the DCA runs shows any performance regressions or changes in alerts. This should be good to go.

**Review by hvitved — APPROVED** [review]

**Pull request merged**